hazelcast/src/test/java/com/hazelcast/genericrecord/AbstractGenericRecordTest.java

```diff
@@ -251,24 +252,90 @@ public void testInconsistentClassDefinition() {
                 new ClassDefinitionBuilder(TestSerializationConstants.PORTABLE_FACTORY_ID, TestSerializationConstants.NAMED_PORTABLE)
                         .addStringField("WrongName").addIntField("myint").build();
 
-
         GenericRecord namedRecord = GenericRecordBuilder.portable(namedPortableClassDefinition)
-                                                        .setString("name", "foo")
-                                                        .setInt("myint", 123).build();
-
+                .setString("name", "foo")
+                .setInt("myint", 123).build();
 
         GenericRecord inConsistentNamedRecord = GenericRecordBuilder.portable(inConsistentNamedPortableClassDefinition)
-                                                                    .setString("WrongName", "foo")
-                                                                    .setInt("myint", 123).build();
-
+                .setString("WrongName", "foo")
+                .setInt("myint", 123).build();
 
         HazelcastInstance instance = createAccessorInstance(serializationConfig);
         IMap<Object, Object> map = instance.getMap("test");
         map.put(1, namedRecord);
+        try {
+            map.put(2, inConsistentNamedRecord);
+            fail("Client should reject incompatible class definition registration attempts");
+        } catch (HazelcastSerializationException ignored) {
+        }
 
+        // When the checkClassDefErrors is set to false, we should not be
```

Comment:
can you separate this part into another test case and keep the unit test simple?

Reply:
Done